### PR TITLE
poll txn status - backend flow implementation

### PR DESCRIPTION
**Pull request by viv-defy**

### CODE REVIEW

File: bundler/src/provider/listeners.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    let success_param = log
+        .params
+        .into_iter()
+        .find(|param| param.name == "success")
+        .unwrap();
+    let success = success_param.value.into_bool().unwrap();
+
+    let status = if success {
+        "success".to_string()
```
Comment (ska1296) [inline bundler/src/provider/listeners.rs:48]: We have an enum of "status". Use "success"/"failure" from there.

**Review by ska1296 — CHANGES_REQUESTED** [review]

File: bundler/src/provider/listeners.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    } else {
+        "failed".to_string()
+    };
+
+    transaction_dao
+        .update_user_transactions(txn_id, txn_hash.to_string(), status)
```
Comment (vishnukvnvv) [inline bundler/src/provider/listeners.rs:54]:
```suggestion
        .update_user_transactions(txn_id, txn_hash, status)
```

**Review by vishnukvnvv — CHANGES_REQUESTED** [review]

File: bundler/src/db/dao/transaction_dao.rs [hunk 3]
```diff
@@ -107,6 +107,14 @@ impl TransactionDao {
             exponent: exponent?,
         })
     }
+
+    pub async fn update_user_transactions(&self, txn_id: String, txn_hash: String, status: String) {
```
Comment (ska1296) [inline bundler/src/db/dao/transaction_dao.rs:111]:
```suggestion
    pub async fn update_user_transaction(&self, txn_id: String, txn_hash: String, status: String) {
```
single txn.

File: bundler/src/db/dao/transaction_dao.rs [hunk 4]
```diff
@@ -107,6 +107,14 @@ impl TransactionDao {
...
+        let conn = connect(self.pool.clone()).await;
+        let mut stmt = conn
+            .prepare("UPDATE user_transactions set status=?, metadata = json_set(metadata, '$.transaction_hash', ?) where transaction_id=?")
```
Comment (ska1296) [inline bundler/src/db/dao/transaction_dao.rs:114]:
```suggestion
            .prepare("UPDATE user_transactions set status = ?, metadata = json_set(metadata, '$.transaction_hash', ?) where transaction_id = ?")
```

consistent spacing helps

**Review by ska1296 — CHANGES_REQUESTED** [review]

**Review by ska1296 — APPROVED** [review]

**Review by vishnukvnvv — APPROVED** [review]

**Pull request merged**